reactor-core/src/main/java/reactor/core/publisher/CallSiteSupplierFactory.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2023 VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import sun.misc.JavaLangAccess;
+import sun.misc.SharedSecrets;
+
+import static reactor.core.publisher.Traces.full;
+import static reactor.core.publisher.Traces.isUserCode;
+import static reactor.core.publisher.Traces.shouldSanitize;
+
+/**
+ * Utility class for the call-site extracting on Java 8.
+ */
+class CallSiteSupplierFactory implements Supplier<Supplier<String>> {
+
+	static final Supplier<Supplier<String>> supplier;
+
+	static {
+		String[] strategyClasses = {
+				CallSiteSupplierFactory.class.getName() + "$SharedSecretsCallSiteSupplierFactory",
+				CallSiteSupplierFactory.class.getName() + "$ExceptionCallSiteSupplierFactory",
+		};
+		// tries to use stack trace traversing approach via and sun.misc.JavaLangAccess
+		// .getStackTrace* or fallbacks to default java.lang.Throwable.getStackTrace
+		// fallbacks to default s
```

Comment:
Please check because it seems some parts are missing and some are duplicated

Reply:
fixed